### ISSUE TITLE
Provider download fails

### ISSUE THREAD

**Issue by BehnazMoradabadi** [terraform-provider-openstack/terraform-provider-openstack#1234]
### Terraform Version
v0.12.18

### Affected Resource(s)
There is an error in downloading openstack provider version `1.41.0`

### Expected Behavior
Terraform init should download the provider but it does not.

### Actual Behavior
I am having this config 
```
provider "openstack" {
  version                       = "~> 1.25"
  auth_url                      = var.auth_url
  tenant_id                     = var.project_id
  tenant_name                   = var.project_name
  application_credential_id     = var.user_name
  application_credential_secret = var.password
  region                        = var.region
}
```
then since 2 days ago after running `terraform init` I get this error:

```
Initializing the backend...

Initializing provider plugins...
- Checking for available provider plugins...

Error verifying GPG signature for provider "openstack"
Terraform was unable to verify the GPG signature of the downloaded provider
files using the keys downloaded from the Terraform Registry. This may mean that
the publisher of the provider removed the key it was signed with, or that the
distributed files were changed after this version was released.

- Downloading plugin for provider "random" (hashicorp/random) 3.1.0...

Warning: registry.terraform.io: For users on Terraform 0.13 or greater, this provider has moved to terraform-provider-openstack/openstack. Please update your source in required_providers.



Error: unable to verify signature

```

### Steps to Reproduce
Please list the steps required to reproduce the issue, for example:+-
1. `terraform init`

For now, I fixed it using this:

```
provider "openstack" {
  version                       = "1.40.0"
  auth_url                      = var.auth_url
  tenant_id                     = var.project_id
  tenant_name                   = var.project_name
  application_credential_id     = var.user_name
  application_credential_secret = var.password
  region                        = var.region
}
```

**Comment by mharm001** [comment]
I am also having the same issue 
while
 `version = "~> 1.40.0"`
 works 
`version = "~> 1.41.0"`
fails with the reported error

**Comment by tsellers-r7** [comment]
You will likely have to update your version of terraform to 0.12.31.  The Terraform GPG signing keys were exposed here recently and they replaced them and released new binaries.

I had a similar issue and switching to the newly released 0.12.31 corrected this for me.


Reference: 
Event
- https://discuss.hashicorp.com/t/hcsec-2021-12-codecov-security-event-and-hashicorp-gpg-key-exposure/23512

Terraform update details:
https://discuss.hashicorp.com/t/terraform-updates-for-hcsec-2021-12/23570
